Mathlib/Geometry/Manifold/Algebra/SmoothGerm.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2023 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+import Mathlib.Geometry.Manifold.Algebra.SmoothFunctions
+import Mathlib.Order.Filter.Germ
+
+/-! # Germs of smooth functions
+
+Germs of smooth functions between manifolds.
+
+## Main definitions and results
+* `smoothGerm I I' N x`: the set of germs of smooth functions `f : M → N` at `x : M`
+* `smoothGerm.toSubring` and friends: if `R` is a smooth ring,
+the space of germs of smooth functions `M → R` is a subring of `Germ (𝓝 x) R`
+
+## Tags
+germ, smooth function, manifold
+
+-/
+
+noncomputable section
+
+open Filter Set
+open scoped Manifold Topology
+
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  -- Declare a smooth manifold `M` over the pair `(E, H)` with model `I`.
+  {E : Type*}
+  [NormedAddCommGroup E] [NormedSpace 𝕜 E] {H : Type*} [TopologicalSpace H]
+  (I : ModelWithCorners 𝕜 E H) {M : Type*} [TopologicalSpace M] [ChartedSpace H M]
+  [SmoothManifoldWithCorners I M]
+  -- Declare a smooth manifold `N` over the pair `(E', H')` with model `I'`.
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E']
+  {H' : Type*} [TopologicalSpace H'] (I' : ModelWithCorners 𝕜 E' H')
+  {N : Type*} [TopologicalSpace N] [ChartedSpace H' N]
+
+namespace smoothGerm
+
+/-! Definition of germs of smooth maps, between any two manifolds. -/
+section definition
+
+variable (N) in
+/-- The set of all germs of smooth functions `M → N` at `x : N`. -/
+def _root_.smoothGerm (x : M) : Set (Germ (𝓝 x) N) :=
+  { Filter.Germ.ofFun f | f : SmoothMap I I' M N }
+
+instance (x : M) [ChartedSpace H' N] : Coe C^∞⟮I, M; I', N⟯ (smoothGerm I I' N x) :=
+  ⟨fun f ↦ ⟨(f : M → N), ⟨f, rfl⟩⟩⟩
+
+@[simp]
+theorem coe_coe (f : C^∞⟮I, M; I', N⟯) (x : M) :
+    ((f : smoothGerm I I' N x) : (𝓝 x).Germ N) = (f : (𝓝 x).Germ N) :=
+  rfl
+
+theorem coe_eq_coe (f g : C^∞⟮I, M; I', N⟯) {x : M} (h : ∀ᶠ y in 𝓝 x, f y = g y) :
+    (f : smoothGerm I I' N x) = (g : smoothGerm I I' N x) := by
+  ext
+  rwa [Germ.coe_eq]
+
+@[simp]
+lemma mem_smoothGerm {x : M} (a : Germ (𝓝 x) N) :
+    a ∈ smoothGerm I I' N x ↔ ∃ f : SmoothMap I I' M N, Germ.ofFun f = a := by
+  rfl
```

Comment:
... and this one should be declared before `namespace smoothGerm` as they are about `smoothGerm` as a set, not a type